lib/rpcenc/reader.go

```diff
@@ -40,7 +41,63 @@ type ReaderStream struct {
 	Info string
 }
 
+var client = func() *http.Client {
+	c := *http.DefaultClient
+	c.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
```

Comment:
It's meant to, we want to handle the redirect ourselves in https://github.com/filecoin-project/lotus/pull/6952/files#diff-309dd201d9384284851cf021bce98a146c76f5db8a982949b87f1e4b3d63cb26R138 (we need to save the last URL form the HEAD requests for the final POST. Unfortunately we can't just do a POST and have Go handle the redirects, because Go's http client will send the request body with the first request, avan if it then gets the redirect response - which is why we do this whole HEAD request dance here)

Reply:
(and we have to use this weird CheckRedirect to be able to save the last redirect `Location` header, as otherwise if you just let the http client handle the redirect, there is no way to get that URL)

Reply:
Note that this error has a very special meaning, perhaps not obvious:
```
ErrUseLastResponse can be returned by Client.CheckRedirect hooks to control how redirects are processed. If returned, the next request is not sent and the most recent response is returned with its body unclosed
```